.github/workflows/bench.yml

```diff
@@ -5,26 +5,40 @@ on:
     # Runs at 08:00 UTC every day
     - cron: "0 8 * * *"
   workflow_dispatch:
+    inputs:
+      max_parallel:
+        description: 'Maximum number of parallel jobs (1-20, use 1 for sequential execution)'
+        required: true
+        default: '10'
+        type: string
 
 jobs:
-  run-benchmarks:
+  pre-warm:
     runs-on: ubuntu-latest
+    steps:
+      - name: Pre-warm audio instance
+        run: curl -f -X POST "https://ai-benchmarks.fly.dev/bench?mode=audio&max_tokens=20&num_requests=1"
 
+  run-benchmarks:
+    needs: pre-warm
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        region: [sea, iad, cdg, nrt]
```

Comment:
I think it's fine to parallelize region since those are separate runners, but parallelizing medium will put additional load on the runners which we know can affect the measurements. I'd suggest either keeping each medium as a serial task or having a separate max-parallel config that we can use to experimentally roll this out.

Reply:
Also, I turned up in Singapore (sin) vs Narita for broader Asia coverage.